test/utils_test.js

```diff
@@ -306,6 +307,210 @@ describe("utils", () => {
         });
       });
 
+      it("should merge schema array item defaults from grandparent for overlapping default definitions", () => {
+        const schema = {
+          type: "object",
+          default: {
+            level1: { level2: ["root-default-1", "root-default-2"] },
+          },
+          properties: {
+            level1: {
+              type: "object",
+              properties: {
+                level2: {
+                  type: "array",
+                  items: [
+                    {
+                      type: "string",
+                      default: "child-default-1",
+                    },
+                    {
+                      type: "string",
```

Comment:
What if I _don't_ want the default to be set for this item to "root-default-2" (but still want to keep the grandparent default the same as it is)? Might there be an intuitive way to "unset" a default? (this is probably beyond the scope of this PR, but could be added in the future)